tests/unit/test_results.py

```diff
@@ -41,6 +42,48 @@ def _get_vcr(self, **kwargs):
             "Set-Cookie",
             "User-Agent",
         ]
+        myvcr.filter_query_parameters = [
+            ("client_id", "foo"),
+        ]
+
+        def redact_key_values(keys_to_redact):
+            def before_record_response(response):
+                # Only do this if the response has not been recorded
+                string_body = response["body"]["string"].decode("utf8")
+                if "REDACTED" not in string_body:
+                    # Only do this is if the body contains one or more
+                    # of the keys to redact.
+                    if any(key in string_body for key in keys_to_redact):
+                        try:
+                            # Marshall into json object, if it is a JSON object.
+                            payload = json.loads(string_body)
+                            if isinstance(payload, list):
+                                payload = payload[0]
+                            for key in keys_to_redact:
+                                if key in payload:
+                                    # Redact the key value
+                                    payload[key] = "REDACTED"
+                            # Write out the updated json object to the response
+                            # body string.
+                            response["body"]["string"] = json.dumps(payload).encode()
+                        except ValueError:
+                            # If it is not a json object, return
+                            return response
```

Comment:
If you `import contextlib`, you can use the `suppress` context manager to simplify things a bit. In this case, if a `JSONDecodeError` is raised, the context manager is simply exited and execution continues at the line following the context manager block. In this case, it's the subsequent `return response` line, which is what you want to do anyway.

Note that I replaced `ValueError` with the more specific `JSONDecodeError` to avoid potentially suppressing a `ValueError` that might signify a bug.

Also, I removed the comments, as they essentially simply repeated what is fairly obvious from the code itself.

```suggestion
                        with contextlib.supress(json.JSONDecodeError):
                            payload = json.loads(string_body)
                            if isinstance(payload, list):
                                payload = payload[0]
                            for key in keys_to_redact:
                                if key in payload:
                                    payload[key] = "REDACTED"
                            response["body"]["string"] = json.dumps(payload).encode()
```